Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -1271,13 +1275,27 @@ public void RegisterNewHyperLink(Hyperlink newHyperlink, string linkUrl)
             try
             {
                 return eventArgs.Handled
-                                ? eventArgs.Image
-                                : new BitmapImage(new Uri(url));
+                    ? eventArgs.Image
+                    : GetImageSource(url);
             }
             catch (Exception)
             {
                 return null;
             }
+
+            ImageSource GetImageSource(string imageUrl)
```

Comment:
Should this take Uri argument?

Reply:
I agree, since you are creating a New Uri twice anyway.